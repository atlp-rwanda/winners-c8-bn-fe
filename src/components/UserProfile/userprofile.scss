.home {
    display: flex;
    background: #F5F5F5;
  
    .homeContainer {
      flex: 6;
  
      .titleDashboard{
        width: 95%;
        margin: 5px auto;
        display: flex;
        justify-content: space-between;
        align-items: center;

        .text{
          display: flex;
          align-items: center;
          padding: 20px;
          margin-top: 10px;
          gap: 20px;
          h3, h4,h6{
            color:#395B64;
            font-size: medium
            ;
          }
          p{
            color: #606060;
            font-style: normal;
            font-weight: 500;
            font-size: 13px;
            line-height: 21px;
          }
        }

        h4{
          color: #606060;
          font-style: normal;
          font-weight: 500;
          font-size: 14px;
        }
      }
      .mainContent{
        width: 95%;
        margin: 0px auto;

        p{
          color: #606060;
          font-style: normal;
          font-weight: 500;
          font-size: 16px;
          line-height: 30px;
        }
        .Content{
          background: #FFFFFF;
          border-radius: 9px;
          padding: 10px 60px;

          p{
            color: #606060;
            font-style: normal;
            font-weight: 400;
            font-size: 16px;
            line-height: 30px;
          }

          .profile_image{
            display: flex;
            align-items: center;
            margin-top: 5px;

            .image_info{
              display: flex;
              align-items: center;

              .avatar{
                width: 91px;
                height: 91px;
                border-radius: 50%;
              }
              .text_info{
                margin-left: 20px;
                h4{
                  font-style: normal;
                  font-weight: 400;
                  font-size: 14px;
                  line-height: 21px;
                }
                p{
                  margin-top: 4px;
                  color: #395B64;
                  font-style: normal;
                  font-weight: 400;
                  font-size: 14px;
                  line-height: 21px;
                  text-decoration: underline;
                }

              }

            }
            .Text_input{
              border: none;
              background: #FFFFFF;
              color: #395B64;
              font-size: large;
              font-weight: 500;
            }
            .Text_input:focus{
              border: none;
              border-color: inherit;
              outline: 0;
              box-shadow: none;
            }
            button{
              margin-left: 80px;
              border: none;
              padding: 8px 8px;
              background: #395B64;
              border-radius: 5px;
              color: #FFFFFF;

            }
            // .file_upload{
            //   color: #FFFFFF;
            //   height: 60px;
            //   width: 240px;
            //   background-color: #395B64;
            //   position: relative;
            //   margin: auto;
            //   right: 0;
            // }
          }
          hr{
            margin-top: 15px;
            margin-bottom: 15px;
            color: #606060;
            opacity: 30%;
          }

          form{
            width: 100%;

            .form_container{
              display: flex;
              align-items: center;
              justify-content: space-between;
              margin-top: 15px;
              
              .form{
                flex-basis: 50%;

                label{
                  color: #606060;
                  font-style: normal;
                  font-weight: 400;
                  font-size: 14px;
                  line-height: 21px;
                }

                input{
                  width: 95%;
                  padding: 5px 2px;
                }
              }
            }
            button{
              border: none;
              padding: 8px 8px;
              background: #395B64;
              border-radius: 5px;
              color: #FFFFFF;
              margin-top: 10px;
              width: 80px;
            }
            button:hover{
              background-color: #508695;
            }
            .update_btn{
              margin-left: 20px;

            }
          }
        }

      }
    }
  }