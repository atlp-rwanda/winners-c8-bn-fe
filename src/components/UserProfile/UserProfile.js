import './userprofile.scss';
import React, { useEffect, useState } from 'react';
import Navbar from '../Navbar/Navbar';
import Sidebar from '../Sidebar/Sidebar';
import { useDispatch, useSelector } from 'react-redux';
import {
  fetchUserProfile,
  updateUserProfile,
} from '../../redux/actions/userProfileAction';
import { rememberInfo } from '../../redux/actions/rememberInfoAction';
import { Skeleton } from '@mui/material';
import { ToastContainer, toast } from 'react-toastify';
import { successToast, errorToast } from '../../helpers/generateToast';
import 'react-toastify/dist/ReactToastify.css';
const UserProfile = () => {
  const userData = useSelector((state) => state.userProfile?.user?.user);
  const [dataForm, setFormData] = useState();
  const [isLoading, setIsLoading] = useState(false);
  const [disable, setDisable] = useState(true);
  const [preview, setPreview] = useState('');
  const [remember, setRemember ] = useState(null);
  const dispatch = useDispatch();

  useEffect(() => {
    fetchUserProfile()(dispatch);
    successToast('Your Profile is loading');
  }, []);

  let formData = new FormData();

  const updateUser = (e) => {
    e.preventDefault();
    successToast('Updating your profile');
    formData.append('firstName', dataForm?.firstName);
    formData.append('lastName', dataForm?.lastName);
    formData.append('department', dataForm?.department);
    formData.append('gender', dataForm?.gender);
    formData.append('image', dataForm?.image);
    formData.append('phoneNumber', dataForm?.phoneNumber);
    formData.append('preferredCurrency', dataForm?.preferredCurrency);
    formData.append('preferredLanguage', dataForm?.preferredLanguage);
    formData.append('username', dataForm?.username);

    updateUserProfile(formData)(dispatch);
    // location.reload();
  };


  useEffect(() => {
    setFormData({
      firstName: userData?.firstName,
      lastName: userData?.lastName,
      email: userData?.email,
      department: userData?.department,
      gender: userData?.gender,
      image: userData?.image,
      phoneNumber: userData?.phoneNumber,
      preferredCurrency: userData?.preferredCurrency,
      preferredLanguage: userData?.preferredLanguage,
      username: userData?.username,
    });
    setIsLoading(true);
  }, [userData]);

  

  //Preview image before upload
  const previewImage = (e) => {
    e.preventDefault();
    setFormData({
      ...dataForm,
      image: e.target.files[0],
    });
    setPreview(URL.createObjectURL(e.target.files[0]));
  };

  const editUser = async (event) => {
    event.preventDefault();
    setDisable(false);
  };



const rememberMe = (e) => {
  e.preventDefault()
  successToast('remembering your information');
  
  dispatch(rememberInfo())
}



  return (
    <>
      <div className="titleDashboard">
        <div className="text">
          <h3>Dashboard&gt;&gt;</h3>
          <h4>Settings&gt;&gt;</h4>
          <h6 data-testid="update-1">Personal information</h6>
        </div>
        <h4>Dashboard&gt;&gt;</h4>
      </div>
      <div className="mainContent">
        <p>Personal information</p>
        {isLoading ? (
          <div className="Content">
            <p>Your photo</p>

            <form action="">
              <div className="profile_image">
                <div className="image_info">
                  <img
                    src={
                      preview
                        ? preview
                        : dataForm?.image
                        ? dataForm?.image
                        : `https://st.depositphotos.com/2101611/3925/v/600/depositphotos_39258143-stock-illustration-businessman-avatar-profile-picture.jpg`
                    }
                    alt=""
                    className="avatar"
                  />

                  <div className="text_info">
                    <input
                      type="text"
                      disabled={disable}
                      defaultValue={dataForm?.username}
                      onChange={(event) =>
                        setFormData({
                          ...dataForm,
                          username: event.target.value,
                        })
                      }
                      className="Text_input"
                    />
                    <p>{dataForm?.email}</p>
                  </div>
                </div>
                <input
                  type="file"
                  name="file_upload"
                  disabled={disable}
                  onChange={(e) => previewImage(e)}
                />
              </div>

              <hr />

              <div className="form_container">
                <div className="form">
                  <label htmlFor="">First Name</label>
                  <br />
                  <input
                    type="text"
                    defaultValue={dataForm?.firstName}
                    disabled={disable}
                    onChange={(event) =>
                      setFormData({
                        ...dataForm,
                        firstName: event.target.value,
                      })
                    }
                  />
                </div>
                <div className="form">
                  <label htmlFor="">Last Name</label>
                  <br />
                  <input
                    type="text"
                    disabled={disable}
                    defaultValue={dataForm?.lastName}
                    onChange={(event) =>
                      setFormData({
                        ...dataForm,
                        lastName: event.target.value,
                      })
                    }
                  />
                </div>
              </div>

              <div className="form_container">
                <div className="form">
                  <label htmlFor="">Title</label>
                  <br />
                  <input
                    type="text"
                    defaultValue={dataForm?.firstName}
                    disabled={disable}
                    onChange={(event) =>
                      setFormData({
                        ...dataForm,
                        firstName: event.target.value,
                      })
                    }
                  />
                </div>
                <div className="form">
                  <label htmlFor="">Gender</label>
                  <br />
                  <input
                    type="text"
                    disabled={disable}
                    defaultValue={dataForm?.gender}
                    onChange={(event) =>
                      setFormData({
                        ...dataForm,
                        gender: event.target.value,
                      })
                    }
                  />
                </div>
              </div>

              <div className="form_container">
                <div className="form">
                  <label htmlFor="">Phone Number</label>
                  <br />
                  <input
                    type="text"
                    disabled={disable}
                    defaultValue={dataForm?.phoneNumber}
                    onChange={(event) =>
                      setFormData({
                        ...dataForm,
                        phoneNumber: event.target.value,
                      })
                    }
                  />
                </div>
                <div className="form">
                  <label htmlFor="">Currency</label>
                  <br />
                  <input
                    type="text"
                    disabled={disable}
                    defaultValue={dataForm?.preferredCurrency}
                    onChange={(event) =>
                      setFormData({
                        ...dataForm,
                        preferredCurrency: event.target.value,
                      })
                    }
                  />
                </div>
              </div>

              <div className="form_container">
                <div className="form">
                  <label htmlFor="">Language</label>
                  <br />
                  <input
                    type="text"
                    disabled={disable}
                    defaultValue={dataForm?.preferredLanguage}
                    onChange={(event) =>
                      setFormData({
                        ...dataForm,
                        preferredLanguage: event.target.value,
                      })
                    }
                  />
                </div>
                <div className="form">
                  <label htmlFor="">Department</label>
                  <br />
                  <input
                    type="text"
                    disabled={disable}
                    defaultValue={dataForm?.department}
                    onChange={(event) =>
                      setFormData({
                        ...dataForm,
                        department: event.target.value,
                      })
                    }
                  />
                </div>
              </div>
              <button type="submit" onClick={editUser}>
                Edit
              </button>
              <button
                type="submit"
                disabled={disable}
                onClick={updateUser}
                className="update_btn"
              >
                Update
              </button>
              <input 
              type='checkbox'
              disabled={disable}
              onChange={rememberMe}
              // onClick={rememberMe}
              className='checkbox'
              />
              <div className='remember'> <p>Remember me</p> </div>

              <ToastContainer />
            </form>
          </div>
        ) : (
          <Skeleton
            variant="rectangle"
            animation="wave"
            width={1100}
            height={800}
          />
        )}
      </div>
    </>
  );
};

export default UserProfile;
