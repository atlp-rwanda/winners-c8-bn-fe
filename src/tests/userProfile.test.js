import React from 'react';
import UserProfile from '../components/UserProfile/UserProfile';
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import thunk from 'redux-thunk';
import rootReducer from '../redux/reducers';
import configureMockStore from 'redux-mock-store';
const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const initialState = {
  userProfile: {
    requestSent: false,
    responseData: {
      isSuccess: undefined,
      message: undefined,
    },
  },
};

let store;
beforeEach(() => {
  store = mockStore(initialState);
});

afterEach(() => {
  cleanup;
});
test('should render user update component', () => {
  render(
    <Provider store={store}>
      <Router>
        <UserProfile />
      </Router>
    </Provider>
  );

  const userUpdateElement = screen.getByTestId('update-1');
  expect(userUpdateElement).toBeInTheDocument();
  expect(userUpdateElement).toHaveTextContent('Personal information');
});

test('<UserUpdate /> matches snapshot', () => {
  const component = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <UserProfile />
        </MemoryRouter>
      </Provider>
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
