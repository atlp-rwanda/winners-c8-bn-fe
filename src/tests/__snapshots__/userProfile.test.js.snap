// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserUpdate /> matches snapshot 1`] = `
Array [
  <div
    className="titleDashboard"
  >
    <div
      className="text"
    >
      <h3>
        Dashboard&gt;&gt;
      </h3>
      <h4>
        Settings&gt;&gt;
      </h4>
      <h6>
        Personal information
      </h6>
    </div>
    <h4>
      Dashboard&gt;&gt;
    </h4>
  </div>,
  <div
    className="mainContent"
  >
    <p>
      Personal information
    </p>
    <span
      className="MuiSkeleton-root MuiSkeleton-rectangle MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
      style={
        Object {
          "height": 800,
          "width": 1100,
        }
      }
    />
  </div>,
]
`;
